### DISCUSSION TITLE
Location problem of QskSubDialogWindow and its subclasses

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#492] [Q&A] [answered]
Why can't QskSubDialogWindow and its subclasses set the position and size through setPlacementPolicy(ignore) and setGeomtry, and can only set the size through setFixedSize? These pop-up windows can only be in the middle of the parent control

**Comment by uwerat** [comment, marked as answer]
Setting the geometries manually is in general no good idea - it looks being simpler in the beginning, but you will pay later. Sizes usually depend on languages, font sizes, icons and what type of content you have.

That's why layout systems do exist - QSkinny is using one that is quite similar to what you find for QWidgets ( Qt/Quick layouts do not cover many relevant situations ).

Of course it is possible to set geometries manually - as long as you do it in line with the layout system. F.e in case of QskSubDialogWindow you can overload QskDialogSubWindow::updateGeometry and do what seems to be right for your situation.

**Reply by hestrro** [reply]
![image](https://github.com/user-attachments/assets/18556e00-53e4-4e79-9af7-4a84e475f596)
I want to implement a drop-down pop-up box of Qsksimplelistbox

**Reply by hestrro** [reply]
This can also be achieved by placing a listview in qsksubwindow

**Reply by uwerat** [reply]
Then you can derive from QskPopup - or simply use QskMenu.
